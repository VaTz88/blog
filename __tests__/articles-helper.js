const fs = require('fs');
const path = require('path');
const matter = require('gray-matter');

const { getArticlesDirArr } = require('../lib/utils/articles-helper');

const articleDirs = getArticlesDirArr();

const pageIdentifierArr = [];

describe('Build articles', () => {
	it.each(articleDirs)('%s should have README.md', dirName => {
		const hasReadme = fs.existsSync(
			path.resolve(__dirname, '..', dirName, 'README.md'),
		);
		expect(hasReadme).toBe(true);
	});

	it.each(articleDirs)('%s should have meta image', dirName => {
		const hasMetaImage = fs.existsSync(
			path.resolve(__dirname, '..', dirName, 'meta-image.png'),
		);
		expect(hasMetaImage).toBe(true);
	});

	it.each(articleDirs)(
		'%s README.md should have correct gray-matter data',
		dirName => {
			const readmeText = fs.readFileSync(
				path.resolve(__dirname, '..', dirName, 'README.md'),
			);
			const { content: readme, data: indexJson } = matter(
				readmeText.toString(),
				{
					delims: ['<!--', '-->'],
				},
			);
			expect(readme).toBeTruthy();
			expect(indexJson).toEqual(
				expect.objectContaining({
					title: expect.any(String),
					description: expect.any(String),
					meta: expect.any(Array),
					date: expect.any(String),
					page_identifier: expect.any(String),
				}),
			);
			// page_identifier should never change for article
			expect(indexJson.page_identifier).toMatchSnapshot();
			pageIdentifierArr.push(indexJson.page_identifier);
		},
	);

	it(`should have unique page_identifier for each article page`, () => {
		const uniques = new Set(pageIdentifierArr);
		expect(pageIdentifierArr.length).toBe(uniques.size);
	});
});
