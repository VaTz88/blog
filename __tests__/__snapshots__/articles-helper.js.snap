// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Build articles all-you-need-to-Know-about-placements-in-vit-vellore README.md should have correct gray-matter data 1`] = `"blog002"`;

exports[`Build articles components-with-async-friendly-event-handlers README.md should have correct gray-matter data 1`] = `"blog004"`;

exports[`Build articles setting-up-ubuntu-on-asus-rog-gl552vw README.md should have correct gray-matter data 1`] = `"blog001"`;

exports[`Build articles upgrading-ejected-cra-project README.md should have correct gray-matter data 1`] = `"blog003"`;
